proposals/records-wip.md

```diff
@@ -67,8 +67,10 @@ Record types produce synthesized implementations for the following methods:
 * `object.Equals(object)` override, unless it is sealed or user provided
 * `T Equals(T)` method, where `T` is the current type
 
-`T Equals(T)` is specified to perform value equality, comparing the property with same name as
-each primary constructor parameter to the corresponding property of the other type.
+`T Equals(T)` is specified to perform value equality such that `Equals` is
+true if and only if all the instance fields of the receiver type are equal
```

Comment:
>equal [](start = 69, length = 5)

I don't know what `equals` means.  How is this possible, given that the fields of the base class might not be accessible?

Reply:
This is the exact same language the spec uses for anonymous types, so if it's underspecified, that's an existing problem :)

Reply:
No, anonymous types don't have a base class. In any case, some other existing text being underspecified isn't an excuse. I think you should just say `object.Equals(x, y)` (and mention as an aside that the implied boxing may be elided)

---
In reply to: [418282509](https://github.com/dotnet/csharplang/pull/3396#discussion_r418282509) [](ancestors = 418282509)